sklearn/ensemble/gradient_boosting.py

```diff
@@ -1421,7 +1421,9 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             self.init_.fit(X, y, sample_weight)
 
             # init predictions
-            y_pred = self.init_.predict(X)
+            y_pred = self.init_.predict(X).astype(np.float64, copy=False)
```

Comment:
What happens if you don't cast it to 64bit?

Reply:
An error is raised due to unsafe casting because y_pred is updated later by the actual fit of the gradient boosting, and filled with float64 values.
```python
y_pred[:, k] += (learning_rate * tree.value[:, 0, 0].take(terminal_regions, axis=0))
```